test/resources/deployment.go

```diff
@@ -195,6 +197,51 @@ func (t *TestDeployment) DeletePodAndWait() {
 	}
 }
 
+func (t *TestDeployment) ForceDeletePod(pod string) error {
+	err := t.Client.CoreV1().Pods(t.Deployment.Namespace).Delete(context.Background(), pod, *metav1.NewDeleteOptions(0))
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	// remove pod from the deployment's pod list
+	pods := make([]PodDetails, len(t.Pods)-1)
+	i := 0
+	for _, tPod := range t.Pods {
+		if tPod.Name == pod {
+			continue
+		}
+		pods[i] = tPod
+		i++
+	}
+	t.Pods = pods
+	return nil
+}
+
+func (t *TestDeployment) WaitForPodTerminating(timeout time.Duration) {
```

Comment:
Can we refactor and merge this method with `DeletePodAndWait` because there seems to be some duplicate logic here.

Reply:
I don't see a huge overlap between the two functions other than the usage of channels to collect results for each goroutine dedicated to each pod. I guess we can refactor them using a generic helper function that takes the wait condition function as the parameter.

Reply:
I was thinking we can refactor the waiting part from `DeletePodAndWait` as a separate method here and add the new stopping conditions we need in this method to `WaitForPodNoLongerRunningInNamespace` or rewrite waiting logic in `DeletePodAndWait` to use what we have here. 
Otherwise, we now have 2 separate functions that wait for deletion in 2 different ways.

Reply:
Unfortunately we cannot modify `WaitForPodNoLongerRunningInNamespace` as it a function from k8s e2e framework package. And it would be the best that we stick with the functions provided via this package when we can for testing any e2e functionality. On that note, I think the best way to refactor this would be to have a generic helper function as suggested above.